mux.go

```diff
@@ -317,15 +312,37 @@ func (mx *Mux) Mount(pattern string, handler http.Handler) {
 	}
 }
 
+// Routes returns a slice of routing information from the tree,
+// useful for traversing available routes of a router.
+func (mx *Mux) Routes() []Route {
+	return mx.tree.routes()
+}
+
 // Middlewares returns a slice of middleware handler functions.
 func (mx *Mux) Middlewares() Middlewares {
 	return mx.middlewares
 }
 
-// Routes returns a slice of routing information from the tree,
-// useful for traversing available routes of a router.
-func (mx *Mux) Routes() []Route {
-	return mx.tree.routes()
+// Find searches the routing tree for a handler that matches the method/path.
+// It's similar to routing a http request, but without executing the handler
+// thereafter.
+//
+// Note: the *Context state is updated during execution, so manage
+// the state carefully or make a NewRouteContext().
+func (mx *Mux) Find(rctx *Context, method, path string) http.Handler {
```

Comment:
Can we return middlewares as a second argument? Would that be possible?

```go
func (mx *Mux) Find(ctx, method, path) (http.Handler, ...func(http.Handler) http.Handler)

// ie.
handler, middlewares := r.Find(ctx, method, path)
```

Reply:
It's possible to offer the middlewares from the Mux#middlewares (the middleware stack), but, it wouldn't be very useful because if you stack middlewares from multiple subrouters, it wont represent the clear execution path to the endpoint handler.

What would be the use case to return the execution middleware/handler execution path here?

Reply:
Initially, I was thinking about different use cases:

1. Being able to change the routes dynamically. (ie. Call a different handler with some middleware stack applied, based on request/context). But this would much more complex and probably not a good idea.

2. Being able to query Router/Mux for specific route and get the Handler and full Middleware stack. (ie. for documentation purposes: `handler, middlewares := r.Find(method, path)` - with this, I could easily verify that a specific route has some middleware applied). But this is something than can be done in `Walk()` anyway, so probably not a good idea either.